### PR TITLE
fix: [CI-6476]: Changed CodebaseTaskParameters field to ParameterField to support expressions

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Changed CodebaseTaskParameters field to ParameterField to support expressions
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41860)

### CODE REVIEW

**Comment by raghavharness** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java [hunk 1]
```diff
@@ -103,11 +104,13 @@ public TaskRequest obtainTask(
...
-        connectorUtils.getConnectorDetails(AmbianceUtils.getNgAccess(ambiance), stepParameters.getConnectorRef());
+    ConnectorDetails connectorDetails = connectorUtils.getConnectorDetails(AmbianceUtils.getNgAccess(ambiance),
+        RunTimeInputHandler.resolveStringParameterV2(
+            "connectorRef", STEP_TYPE.getType(), STEP_TYPE.getType(), stepParameters.getConnectorRef(), false));
 
-    ScmGitRefTaskParams scmGitRefTaskParams =
-        obtainTaskParameters(manualExecutionSource, connectorDetails, stepParameters.getRepoName());
+    ScmGitRefTaskParams scmGitRefTaskParams = obtainTaskParameters(manualExecutionSource, connectorDetails,
+        RunTimeInputHandler.resolveStringParameterV2(
+            "repoName", STEP_TYPE.getType(), STEP_TYPE.getType(), stepParameters.getRepoName(), false));
```
Comment (aman-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java:113]: @raghavharness Param 3 is stepIdentifier. Why are we using STEP_TYPE.getType() in it?

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java [hunk 2]
```diff
@@ -103,11 +104,13 @@ public TaskRequest obtainTask(
     }
 
     ManualExecutionSource manualExecutionSource = (ManualExecutionSource) executionSource;
-    ConnectorDetails connectorDetails =
-        connectorUtils.getConnectorDetails(AmbianceUtils.getNgAccess(ambiance), stepParameters.getConnectorRef());
+    ConnectorDetails connectorDetails = connectorUtils.getConnectorDetails(AmbianceUtils.getNgAccess(ambiance),
+        RunTimeInputHandler.resolveStringParameterV2(
+            "connectorRef", STEP_TYPE.getType(), STEP_TYPE.getType(), stepParameters.getConnectorRef(), false));
```
Comment (aman-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java:109]: same as above

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java (hunk 1 above)
Comment (raghavharness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java:113]: For codebasetask we don't an identifier as such that's why kept it same.

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java (hunk 2 above)
Comment (raghavharness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseTaskStep.java]: done

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41860#-:-NJtW7UGDjvK5UvdQ4Gh:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**